### ISSUE TITLE
Legacy provisioned alerts are not migrated to Grafana Alerting

### ISSUE THREAD

**Issue by Dgotlieb** [grafana/grafana#43546]
Hi,

After upgrading the Docker image version **8.2.5** to **8.3.2** we no longer see the alerts under Alerting-->Alerting rules:
![image](https://user-images.githubusercontent.com/12662834/147552658-b12cbba6-201b-4ddc-8a0b-fa02fbf172e9.png)

I saw [in the docs](https://grafana.com/docs/grafana/next/alerting/unified-alerting/opt-in/) that after version 8.3, existing legacy dashboard alerts should be migrated automatically.

I also tried explicitly setting `unified_alerting-->enabled-->true` and  `alerting-->enabled-->false` in `grafana.ini` but nothing changed.

There are some error logs but I'm not sure how they are related and what can get done to fix it:
`t=2021-12-28T09:47:48+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/api/ruler/2/api/v1/rules status=500 remote_addr=127.0.0.1 time_ms=5 size=96 referer=http://localhost:3000/alerting/list`

** Not sure it matters but all dashboards are provisioned with file provider.

Thanks

**Comment by mrecamales2222** [comment]
Hi,

I'm following the issue as I have the same problem. Do we have any news on that?

Thanks

**Comment by niamR** [comment]
Hi,

Same here, alerts comming from provisioned dashboards via file do not get migrated.
Tested with target version 9.0.2 and 8.4.3

With legacy ```[alerting] enabled = true``` alerts gets properly provisionned.
But with ```[unified_alerting] enabled = true``` it looks like the alerts didn't get recognized by the migration process:
```logger=migrator t=2022-07-04T15:15:50.317467781+02:00 level=info msg="alerts found to migrate" alerts=0```

**Comment by grobinson-grafana** [comment]
Hi, would it be possible to upgrade to the latest 8.x version of Grafana which is 8.5.6 and confirm whether the issue is still present?

**Comment by Dgotlieb** [comment]
@grobinson-grafana I can confirm that this is still happening in both `8.5.6` and `9.0.2`.
I also noticed the below logs when Grafana is starting:

> logger=migrator t=2022-07-11T10:34:42.84+0000 lvl=info msg="Executing migration" id="managed folder permissions alert actions migration"
logger=migrator t=2022-07-11T10:34:42.84+0000 lvl=info msg="migrations completed" performed=397 skipped=0 duration=2.359274377s
logger=migrator t=2022-07-11T10:32:17.582317471Z level=info msg="Executing migration" id="clear migration entry \"remove unified alerting data\""
logger=migrator t=2022-07-11T10:32:17.582534111Z level=info msg="Executing migration" id="move dashboard alerts to unified alerting"
logger=migrator t=2022-07-11T10:32:17.582762536Z level=info msg="alerts found to migrate" alerts=0

Thanks

**Comment by grobinson-grafana** [comment]
Hi, thanks the log output. Do you see rules in the `alert_rule` table in the database, or are there no rows in that database?

**Comment by Dgotlieb** [comment]
Nope..
<img width="978" alt="image" src="https://user-images.githubusercontent.com/12662834/178971600-5160b289-4e4b-4c36-8bcf-f51dd9940f37.png">

**Comment by Dgotlieb** [comment]
Hi,

1. Your query returns my 80 alerts.
2. All alerts belong to `org: 1` (main organization) which exists in the `org` table
3. I didn't check all 80 alerts but some show up in the `dashboard` table

**Comment by Dgotlieb** [comment]
Yap, I'm also not sure I understand 😓 but this is Grafana internals which I'm not familiar with.
Just to recap, the status is:
`alert` table has 80 entries
`alert_rule` table is empty
`org` table holds one entry - `id: 1` and `name: Main Org.`
`dashboard` table holds 36 dashboards

I noticed in the `alert_rules` tab, that Prometheus alerts are shown while Grafana alerts are empty:
<img width="395" alt="image" src="https://user-images.githubusercontent.com/12662834/179683566-935ebe18-ccb6-48b0-90ce-9d0960b7e292.png">

Thanks

**Comment by armandgrillet** [comment]
Did you try disabling Grafana Alerting, starting it, then reenable Grafana Alerting and start Grafana again. We have updated the migration part quite a bit since this issue exist. This can be done using https://grafana.com/docs/grafana/latest/alerting/migrating-alerts/opt-out/#disable-alerting and https://grafana.com/docs/grafana/latest/alerting/migrating-alerts/opt-in/. 
Could we get the logs of Grafana when the last part is done as well? 
Also: how is the provisioning of alerts done? Migration happens before provisioning, this could be the reason for this issue.

**Comment by armandgrillet** [comment]
After re-reading this, the problem is this:

- When Grafana starts with Grafana Alerting enabled, we migrate the alerts in the old table.
- The legacy alerts are then provisioned using the files
- The provisioned legacy alerts have not been migrated

As legacy alerts are deprecated and will be removed in Grafana 10, we should:
- ASAP a warning in Grafana Alerting stating if provisioned legacy alerts exist, stating that these alerts have not been migrated and link to a doc stating how to do the migration.
- In Grafana 10 we should error at launch if there are provisioned legacy alerts with a link to a doc stating how to do the migration.

**Comment by Dgotlieb** [comment]
In case someone else is stuck with this issue.. I took @armandgrillet suggestion and added another step to it:
1. disable alerting:
``` ini
[alerting]
enabled = false

[unified_alerting]
enabled = false
```
2. enable `unified_alerting` 
``` ini
[alerting]
enabled = false

[unified_alerting]
enabled = true
```
**(at this point alerts still not showing)**

3. enable "legacy" alerting:
```
[alerting]
enabled = true

[unified_alerting]
enabled = false
```

The result is the below warning which I guess I will need to live with until Grafana 10.. hopefully it will be resolved before
![image](https://user-images.githubusercontent.com/12662834/185797047-781af526-54a0-40ca-900f-c4fe69d283af.png)


Perhaps I'm missing something but this is the ONLY way I was able to make it work.

**Comment by armandgrillet** [comment]
@Dgotlieb Yes we will fix this bug before removing legacy alerting. Thank you for reporting this issue, we're still investigating on the best way to fix this.

**Comment by JacobsonMT** [comment]
Closing with the removal of legacy alerting in v11.

The new [preview tool method of upgrading](https://grafana.com/docs/grafana/v10.4/alerting/set-up/migrating-alerts/#upgrade-with-preview-recommended) to Grafana Alerting available in v10.3+ gives the opportunity to upgrade your alerts in a safe preview environment before cutting over to Grafana Alerting. During this preview you can export your provisioned alerts in file-provisioning or HCL format so you can migrate your as-code setup at your leisure.
